### ISSUE TITLE
CPU input validation is not working

### ISSUE THREAD

**Issue by mohamedamer453** [threefoldtech/terraform-provider-grid#313]
## Description

I was testing the input validation for `CPU` in the singlenode example and when i entered `cpu = 9999` the vm was deployed without any validation errors and when i checked the number of cores from the vm after deployment using `lscpu` the number of cores was `15` 

![image](https://user-images.githubusercontent.com/101194226/188436728-630598f0-5fc5-456b-9e66-43d38a9f0928.png)

and i didn't mention anywhere in the config that the cpu value should be `15`

## Deployment Output

and as seen in the deployment output the cpu value was `9999`

https://gist.github.com/mohamedamer453/9dee840f1cce70f8c8896af15ac81b44

## main.tf 

```
terraform {
  required_providers {
    grid = {
      source = "threefoldtech/grid"
    }
  }
}

provider "grid" {
}

resource "grid_network" "net1" {
    nodes = [3]
    ip_range = "10.1.0.0/16"
    name = "network"
    description = "newer network"
    add_wg_access = true
}
resource "grid_deployment" "d1" {
  node = 3
  network_name = grid_network.net1.name
  ip_range = lookup(grid_network.net1.nodes_ip_range, 3, "")
  vms {
    name = "vm1"
    flist = "https://hub.grid.tf/tf-official-apps/threefoldtech-ubuntu-20.04.flist"
    cpu = 9999
    publicip = true
    memory = 1024
    entrypoint = "/init.sh"
    env_vars = {
      SSH_KEY = "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQC533B35CELELtgg2d7Tsi5KelLxR0FYUlrcTmRRQuTNP9arP01JYD8iHKqh6naMbbzR8+M0gdPEeRK4oVqQtEcH1C47vLyRI/4DqahAE2nTW08wtJM5uiIvcQ9H2HMzZ3MXYWWlgyHMgW2QXQxzrRS0NXvsY+4wxe97MMZs9MDs+d+X15DfG6JffjMHydi+4tHB50WmHe5tFscBFxLbgDBUxNGiwi3BQc1nWIuYwMMV1GFwT3ndyLAp19KPkEa/dffiqLdzkgs2qpXtfBhTZ/lFeQRc60DHCMWExr9ySDbavIMuBFylf/ZQeJXm9dFXJN7bBTbflZIIuUMjmrI7cU5eSuZqAj5l+Yb1mLN8ljmKSIM3/tkKbzXNH5AUtRVKTn+aEPvJAEYtserAxAP5pjy6nmegn0UerEE3DWEV2kqDig3aPSNhi9WSCykvG2tz7DIr0UP6qEIWYMC/5OisnSGj8w8dAjyxS9B0Jlx7DEmqPDNBqp8UcwV75Cot8vtIac= root@mohamed-Inspiron-3576"
    }
    planetary = true
  }
  # vms {
  #   name = "anothervm1"
  #   flist = "https://hub.grid.tf/tf-official-apps/base:latest.flist"
  #   cpu = 1
  #   memory = 1024
  #   entrypoint = "/sbin/zinit init"
  #   env_vars = {
  #     SSH_KEY = "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQC533B35CELELtgg2d7Tsi5KelLxR0FYUlrcTmRRQuTNP9arP01JYD8iHKqh6naMbbzR8+M0gdPEeRK4oVqQtEcH1C47vLyRI/4DqahAE2nTW08wtJM5uiIvcQ9H2HMzZ3MXYWWlgyHMgW2QXQxzrRS0NXvsY+4wxe97MMZs9MDs+d+X15DfG6JffjMHydi+4tHB50WmHe5tFscBFxLbgDBUxNGiwi3BQc1nWIuYwMMV1GFwT3ndyLAp19KPkEa/dffiqLdzkgs2qpXtfBhTZ/lFeQRc60DHCMWExr9ySDbavIMuBFylf/ZQeJXm9dFXJN7bBTbflZIIuUMjmrI7cU5eSuZqAj5l+Yb1mLN8ljmKSIM3/tkKbzXNH5AUtRVKTn+aEPvJAEYtserAxAP5pjy6nmegn0UerEE3DWEV2kqDig3aPSNhi9WSCykvG2tz7DIr0UP6qEIWYMC/5OisnSGj8w8dAjyxS9B0Jlx7DEmqPDNBqp8UcwV75Cot8vtIac= root@mohamed-Inspiron-3576"
  #   }
  #   planetary = true
  # }
}
output "wg_config" {
    value = grid_network.net1.access_wg_config
}
output "node1_zmachine1_ip" {
    value = grid_deployment.d1.vms[0].ip
}
# output "node1_zmachine2_ip" {
#     value = grid_deployment.d1.vms[1].ip
# }
output "public_ip" {
    value = grid_deployment.d1.vms[0].computedip
}

output "ygg_ip" {
    value = grid_deployment.d1.vms[0].ygg_ip
}

# output "ygg_ip2" {
#     value = grid_deployment.d1.vms[1].ygg_ip
# }
``

**Comment by MarioBassem** [comment]
https://github.com/threefoldtech/terraform-provider-grid/pull/316

**Comment by khaledyoussef24** [comment]
### verified on devnet

1. tried to deploy a vm with 999 cpus
2. deployment failed
![Screenshot from 2022-09-24 18-08-10](https://user-images.githubusercontent.com/110984055/192108459-99e346cf-7cad-4c7f-a05a-d6eee964c7b3.png)
![Screenshot from 2022-09-24 18-07-20](https://user-images.githubusercontent.com/110984055/192108461-206bd622-5a1b-495f-bd50-6445be9cb80f.png)

**Comment by A-Harby** [comment]
Test steps: [TC445 - Deploy a VM with invalid resources (CPU)](https://app.testlodge.com/a/26076/projects/40893/suites/193850?expand_section=335869#case_3451750)

**Comment by khaledyoussef24** [comment]
### the problem still exist 
**steps to produce**

1. open terraform
2. go to single vm
3. make an invalid CPU and invalid memory
4. it will be deployed giving a random number of resources

![Screenshot from 2022-11-07 08-44-47](https://user-images.githubusercontent.com/110984055/200293079-c0338ee7-c4be-4e06-b755-69b4840c5c89.png)

**Comment by MarioBassem** [comment]
> ### the problem still exist
> **steps to produce**
> 
> 1. open terraform
> 2. go to single vm
> 3. make an invalid CPU and invalid memory
> 4. it will be deployed giving a random number of resources
> 
> ![Screenshot from 2022-11-07 08-44-47](https://user-images.githubusercontent.com/110984055/200293079-c0338ee7-c4be-4e06-b755-69b4840c5c89.png)

fixed by (https://github.com/threefoldtech/terraform-provider-grid/pull/323)
will be released in v1.5.9

**Comment by khaledyoussef24** [comment]
### verified on devnet
**steps to produce**

1. open terraform
2. go to single vm
3. make an invalid CPU and invalid memory
4. run `terraform init && terraform apply -parallelism=1 `
5. the deployment will fail and "CPUs must be more than or equal to 1 and less than or equal to 32: Invalid CPU input" will appear

![Screenshot from 2022-11-23 17-43-48](https://user-images.githubusercontent.com/110984055/203589155-5cf52ed5-dc68-40c7-81a8-23e0f0d93549.png)
![Screenshot from 2022-11-23 17-42-07](https://user-images.githubusercontent.com/110984055/203589160-e4de495c-bb26-45eb-924d-8e93de79f223.png)
